sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -648,7 +648,11 @@ class SparkSession private(
    * @since 2.0.0
    */
   def sql(sqlText: String): DataFrame = {
-    Dataset.ofRows(self, sessionState.sqlParser.parsePlan(sqlText))
+    val tracker = new QueryPlanningTracker
```

Comment:
Hi, @rxin .
Can we have an option to disable this new feature?

Reply:
@dongjoon-hyun just out of curiosity, what would you like to disable here?

Reply:
It's because this is a Nice-To-Have instead of Must-Have. The framework seems to be designed to support disabling by assigning `None` here.

Reply:
Thanks. I got it, @rxin .